src/dataflow/src/source/postgres/metrics.rs

```diff
@@ -7,51 +7,106 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
-use prometheus::core::AtomicU64;
+use std::collections::HashMap;
 
+use mz_dataflow_types::postgres_source::PostgresSourceDetails;
 use mz_expr::SourceInstanceId;
 use mz_ore::metrics::{CounterVecExt, DeleteOnDropCounter, DeleteOnDropGauge, GaugeVecExt};
+use prometheus::core::AtomicU64;
 
 use crate::source::metrics::SourceBaseMetrics;
 
+#[derive(Eq, PartialEq, Ord, PartialOrd, Debug, Hash, Copy, Clone)]
+pub enum PgOp {
+    INSERT,
+    UPDATE,
+    DELETE,
+}
+
+impl From<&PgOp> for String {
+    fn from(op: &PgOp) -> String {
+        match op {
+            PgOp::INSERT => String::from("INSERT"),
+            PgOp::UPDATE => String::from("UPDATE"),
+            PgOp::DELETE => String::from("DELETE"),
+        }
+    }
+}
 pub(super) struct PgSourceMetrics {
-    pub inserts: DeleteOnDropCounter<'static, AtomicU64, Vec<String>>,
-    pub updates: DeleteOnDropCounter<'static, AtomicU64, Vec<String>>,
-    pub deletes: DeleteOnDropCounter<'static, AtomicU64, Vec<String>>,
-    pub ignored: DeleteOnDropCounter<'static, AtomicU64, Vec<String>>,
-    pub total: DeleteOnDropCounter<'static, AtomicU64, Vec<String>>,
-    pub transactions: DeleteOnDropCounter<'static, AtomicU64, Vec<String>>,
-    pub tables: DeleteOnDropGauge<'static, AtomicU64, Vec<String>>,
-    pub lsn: DeleteOnDropGauge<'static, AtomicU64, Vec<String>>,
+    /// Maps table_oid -> operation -> DeleteOnDropCounter with appropriate tag values
+    table_ops: HashMap<u32, HashMap<PgOp, DeleteOnDropCounter<'static, AtomicU64, Vec<String>>>>,
+    ignored: DeleteOnDropCounter<'static, AtomicU64, Vec<String>>,
+    transactions: DeleteOnDropCounter<'static, AtomicU64, Vec<String>>,
+    tables: DeleteOnDropGauge<'static, AtomicU64, Vec<String>>,
+    lsn: DeleteOnDropGauge<'static, AtomicU64, Vec<String>>,
 }
 
 impl PgSourceMetrics {
-    pub(super) fn new(base_metrics: &SourceBaseMetrics, source_id: SourceInstanceId) -> Self {
-        let labels = &[source_id.to_string()];
+    pub(super) fn new(
```

Comment:
do we call this once per `CREATE VIEWS FROM` call? cause thats when we know the tables?